syntax = "proto2";
import public "common_nm.proto";

enum PortSpeed {
    FORCE10MODEFDX = 0;
    FORCE10MODEHDX = 1;
    FORCE100MODEFDX = 2;
    FORCE100MODEHDX = 3;
    FORCE1GMODEFDX = 4;
    AUTONEGMODE = 5;
    FORCE2G5MODEFDX = 6;
    FORCE5GMODEFDX = 7;
    FORCE10GMODEFDX = 8;
    FORCE12GMODEFDX = 9;
}

enum BPDUStates {
    BLOCKING = 0;
    LISTENING = 1;
    LEARNING = 2;
    FORWARDING = 3;
    DISABLED = 4;
}

message ConfigPortState {
    required string namedPort = 1;
    required BPDUStates state = 2;
}

message BPDUPortState {
    required string namedPort = 1;
    required BPDUStates state = 2;
    required bool success = 3;
    optional ErrorMessage error = 4;
}

message PortInfoReq {
    repeated string portInfoKey = 1;
}

message PortInfoResp {
    repeated ValueResp values = 1;
}

message ConfigBPDUReq {
    required uint32 vlanid = 1;
}

message ConfigBPDUResp {
    required bool status = 1;
    optional ErrorMessage error = 2;
}

message ConfigPortStateReq {
    repeated ConfigPortState state = 1;
}

message ConfigPortStateResp{
    repeated BPDUPortState state = 1;
}
